### DISCUSSION TITLE
EllipseShapeData?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1247] [Ideas]
Hi:

We tried to define an Ellipse shape as HitBox shape
but the BoundingShape is sealed 
so is it possible for adding a new shape which is Ellipse Shape 

Thanks

**Comment by AlmasB** [comment]
It is possible to implement, but the underlying check is still done based on bounding boxes (even if the shape is a circle).

This is different for `PhysicsComponent` based entities, for which a more precise algorithm is used based on the actual shape. Is your use case based on entities that use `PhysicsComponent`?

**Reply by chengenzhao** [reply]
Yes, it depends on physics component 
use physics component to determine the collision 
for now we use Circle and Rectangle to detect the collision

**Reply by AlmasB** [reply]
The physics engine is box2d, which does not have an ellipse shape. Here's a possible workaround: https://stackoverflow.com/questions/10032756/how-to-create-ellipse-shapes-in-box2d

`BoundingShape.polygon()` might be useful for that
